ui/src/app/auth/register/register.component.ts

```diff
@@ -0,0 +1,70 @@
+import { animate, keyframes, state, style, transition, trigger } from '@angular/animations';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { MustMatch } from 'src/app/Custom Validator/ConfirmPass.validator';
+import { registerScreenAnimation } from '../register.animation';
+import { routerTransition } from '../router.animation';
+
+@Component({
+  selector: 'app-register',
+  animations: [routerTransition(),registerScreenAnimation()],
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.sass']
+})
+export class RegisterComponent implements OnInit {
+
+  registerForm: FormGroup;
+  formSubmitted: boolean = false;
+  hidePassword = true;
+  hideConfirmPassword = true;
+  registerScreenAnimation: string = 'unchecked';
+
+  constructor(private fb: FormBuilder, private route:Router) { }
+
+  ngOnInit(): void {
+    this.registerForm = this.fb.group({
+      email: ['', [Validators.required, Validators.pattern(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/)]],
+      password: ['', [Validators.required,Validators.pattern(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/)]],
+      confirmPassword: ['', [Validators.required]]
+
+    },
+     { validator: MustMatch('password', 'confirmPassword')});
+  }
+ 
+    get email(){
```

Comment:
where is the using of those functions

Reply:
In order to store the data that the user writes in the database and then call it in the login and compare it with the data entered